sql/catalyst/src/main/scala/org/apache/spark/sql/types/DataTypeConversions.scala

```diff
@@ -20,24 +20,21 @@ package org.apache.spark.sql.types
 import java.text.SimpleDateFormat
 
 import org.apache.spark.sql.Row
-import org.apache.spark.sql.catalyst.ScalaReflection
-import org.apache.spark.sql.catalyst.expressions.GenericMutableRow
+import org.apache.spark.sql.catalyst.CatalystTypeConverters
+import org.apache.spark.sql.catalyst.expressions.GenericRowWithSchema
 
 
 private[sql] object DataTypeConversions {
 
   def productToRow(product: Product, schema: StructType): Row = {
```

Comment:
This is only used from LocalRelation AFAICT, but I think we should also add a comment that this method is slow, and that users should use CatalystTypeConverters.createToCatalystConverter for batch conversions -- we don't want future contributors to rely on such a convenient method without at least exposing the performance characteristics.